client/src/modules/authorization/components/SignUpForm.jsx

```diff
@@ -0,0 +1,309 @@
+      isNameValid,
+      isPasswordConfirmValid,
+      isPasswordValid
+    } = this.state;
+
+    const isError = _find(higherLevelErrors, error => error !== '');
+
+    if (
+      isNameValid &&
+      isEmailValid &&
+      isPasswordValid &&
+      isPasswordConfirmValid &&
+      !isError
+    ) {
+      return this.setState({ isFormValid: true });
+    }
+
+    this.setState({ isFormValid: false });
+  };
+
+  handleSignUp = () => {
+    const { signUp } = this.props;
+    const { email, name, password, passwordConfirm } = this.state;
+
+    this.setState({ pending: true });
+
+    this.pendingPromise = makeAbortablePromise(
+      signUp(email, name, password, passwordConfirm)
+    );
+    this.pendingPromise.promise
+      .then(() => this.setState({ confirmationSend: true, pending: false }))
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          const { status } = err.response;
+          const newState = { pending: false };
+
+          if (status === 406) {
+            const { errors } = err.response;
+            newState.higherLevelErrors = errors;
+          }
+
+          this.setState(newState);
+        }
+      });
+  };
+
+  renderSignUpForm = () => {
+    const {
+      higherLevelErrors: {
+        nameError,
+        emailError,
+        passwordConfirmError,
+        passwordError
+      },
+      isFormValid,
+      pending
+    } = this.state;
+    const { onCancel } = this.props;
+
+    return (
+      <Fragment>
+        <h1 className="Sign-Up-Form__heading">
+          <FormattedMessage id="authorization.create-account" />
+        </h1>
+        <form className="Sign-Up-Form__form" noValidate>
+          <SignUpInput
+            disabled={pending}
+            externalErrorId={nameError}
+            focus
+            labelId="authorization.input.username.label"
+            name="name"
+            onChange={this.onNameChange}
+            type="text"
+            validator={this.nameValidator}
+          />
+          <SignUpInput
+            disabled={pending}
+            externalErrorId={emailError}
+            labelId="authorization.input.email.label"
+            name="email"
+            onChange={this.onEmailChange}
+            type="text"
+            validator={this.emailValidator}
+          />
+          <SignUpInput
+            disabled={pending}
+            externalErrorId={passwordError}
+            labelId="authorization.input.password.label"
+            name="password"
+            onChange={this.onPasswordChange}
+            type="password"
+            validator={this.passwordValidator}
+          />
+          <SignUpInput
+            disabled={pending}
+            externalErrorId={passwordConfirmError}
+            labelId="authorization.input.password.confirm"
+            name="confirm"
+            onChange={this.onPasswordConfirmChange}
+            type="password"
+          />
+          <div className="Sign-Up-Form__buttons">
```

Comment:
do not use uppercase letters in class names.
Please change it everywhere to lower case